packages/rich-text-editor/src/vaadin-rich-text-editor.js

```diff
@@ -593,6 +593,25 @@ class RichTextEditor extends ElementMixin(ThemableMixin(PolymerElement)) {
       });
     });
 
+    const TAB_KEY = 9;
+
+    editorContent.addEventListener('keydown', (e) => {
+      if (e.key === 'Escape') {
+        this.__tabBindings = this._editor.keyboard.bindings[TAB_KEY];
+        this._editor.keyboard.bindings[TAB_KEY] = null;
+      } else if (this.__tabBindings) {
+        this._editor.keyboard.bindings[TAB_KEY] = this.__tabBindings;
+        this.__tabBindings = null;
+      }
+    });
+
+    editorContent.addEventListener('blur', () => {
+      if (this.__tabBindings) {
+        this._editor.keyboard.bindings[TAB_KEY] = this.__tabBindings;
+        this.__tabBindings = null;
+      }
+    });
```

Comment:
This a solution to avoid using an element inside RTE to move the focus to when <kbd>Esc</kbd> is pressed.

I tried adding a custom keyboard binding that would read a flag whenever <kbd>Esc</kbd> is pressed, but it wasn't possible because Quill's default bindings are triggered before the custom one (in my tests, the custom binding didn't even run because Quill use `Array.some` and one default binding returns `true` and the execution stops).
Another alternative is to overwrite the default binding for <kbd>Tab</kbd> by copying the original handler to the new one, but only executing it if <kbd>Esc</kbd> hadn't been pressed before <kbd>Tab</kbd>. But the original handler uses some API that is not exposed, so I had to discard this approach as well.

I am not super happy with the proposed solution, but it makes it possible to have the behavior we want to work in Safari and Firefox as well.